sdk/src/main/scala/com/horizen/account/mempool/MempoolMap.scala

```diff
@@ -0,0 +1,149 @@
+package com.horizen.account.mempool
+
+import com.horizen.SidechainTypes
+import com.horizen.account.transaction.EthereumTransaction
+import scorex.util.{ModifierId, ScorexLogging}
+
+import java.math.BigInteger
+import scala.collection.concurrent.TrieMap
+import scala.collection.mutable
+import scala.util.Try
+
+case class MempoolMap() extends ScorexLogging {
+
+  val all = TrieMap[ModifierId, SidechainTypes#SCAT]()
+  val executableTxs = TrieMap[SidechainTypes#SCP, scala.collection.mutable.SortedMap[BigInteger, ModifierId]]()
+  val nonExecutableTxs = TrieMap[SidechainTypes#SCP, scala.collection.mutable.SortedMap[BigInteger, ModifierId]]()
+  val nonces = TrieMap[SidechainTypes#SCP, BigInteger]()
+
+  def contains(ethTransaction: SidechainTypes#SCAT): Boolean = all.contains(ethTransaction.id)
+
+  def containsAccountInfo(account: SidechainTypes#SCP): Boolean = nonces.contains(account)
+
+  def initializeAccount(stateNonce: BigInteger, account: SidechainTypes#SCP): Try[MempoolMap] = Try {
+    nonces.put(account, stateNonce)
+    this
+  }
+
+  def add(ethTransaction: SidechainTypes#SCAT): Try[MempoolMap] = Try {
+    if (!containsAccountInfo(ethTransaction.getFrom)) {
+      log.error(s"Adding transaction failed because mempoolMap is not initialized for account ${ethTransaction.getFrom}")
+      throw new IllegalStateException(s"MempoolMap is not initialized for account ${ethTransaction.getFrom}")
+    }
+
+    if (!contains(ethTransaction)) {
+
+      val account = ethTransaction.getFrom
+      val expectedNonce = nonces.get(account).get
```

Comment:
Cosmetics: here (and in other similar cases) consider using `nonces(account) `instead of  `nonces.get(account).get`

Reply:
Done